pkg/report/table/table.go

```diff
@@ -64,6 +84,175 @@ func (tw Writer) Write(_ context.Context, report types.Report) error {
+	}
+	t.Render()
+
+	// Show legend
+	if err := tml.Fprintf(tw.Output, "Legend:\n"+
+		"- '-': Not scanned\n"+
+		"- '0': Clean (no security findings detected)\n\n"); err != nil {
+		return err
+	}
+	return nil
+}
+
+// showEmptyResultsWarning shows WARN why the results array is empty based on the enabled scanners.
+// We need to separate the vuln/misconfig and secret/license scanners,
+// because the results array contains results without findings for vulns/misconfig only.
+func (tw Writer) showEmptyResultsWarning() {
+	resultByFiles := []types.Scanner{
+		types.VulnerabilityScanner,
+		types.MisconfigScanner,
+	}
+	resultByFindings := []types.Scanner{
+		types.SecretScanner,
+		types.LicenseScanner,
+	}
+
+	var warnStrings []string
+	if scanners := lo.Intersect(resultByFiles, tw.Scanners); len(scanners) > 0 {
+		warnStrings = append(warnStrings, fmt.Sprintf("Supported files for %s scanner(s) not found.",
+			strings.Join(xstrings.ToStringSlice(scanners), "/")))
+	}
+	if scanners := lo.Intersect(resultByFindings, tw.Scanners); len(scanners) > 0 {
+		warnStrings = append(warnStrings, fmt.Sprintf("No results found for %s scanner(s).",
+			strings.Join(xstrings.ToStringSlice(scanners), "/")))
+	}
+
+	if len(warnStrings) == 0 {
+		warnStrings = append(warnStrings, "Scanners are not enabled.")
+	}
+
+	log.WithPrefix("report").Info(strings.Join(warnStrings, " "))
+}
+
+// splitAggregatedPackages splits aggregated packages into different results with path as target.
+// Other results will be returned as is.
+func splitAggregatedPackages(results types.Results) types.Results {
+	var newResults types.Results
+
+	for _, result := range results {
+		if !slices.Contains(ftypes.AggregatingTypes, result.Type) &&
+			// License results from applications don't have `Type`.
+			(result.Class != types.ClassLicense || !slices.Contains(lo.Values(langpkg.PkgTargets), result.Target)) {
+			newResults = append(newResults, result)
+			continue
+		}
+
+		newResults = append(newResults, splitAggregatedVulns(result)...)
+		newResults = append(newResults, splitAggregatedLicenses(result)...)
+
+	}
+	return newResults
+}
+
+func splitAggregatedVulns(result types.Result) types.Results {
+	var newResults types.Results
+
+	// Save packages to display them in the table even if no vulnerabilities were found
+	vulns := lo.SliceToMap(result.Packages, func(pkg ftypes.Package) (string, []types.DetectedVulnerability) {
+		return rootJarFromPath(pkg.FilePath), []types.DetectedVulnerability{}
+	})
+
+	for _, vuln := range result.Vulnerabilities {
+		pkgPath := rootJarFromPath(vuln.PkgPath)
+		vulns[pkgPath] = append(vulns[pkgPath], vuln)
+	}
+	for pkgPath, v := range vulns {
+		newResult := result
+		newResult.Target = lo.Ternary(pkgPath != "", pkgPath, result.Target)
+		newResult.Vulnerabilities = v
+
+		newResults = append(newResults, newResult)
+	}
+
+	sort.Slice(newResults, func(i, j int) bool {
+		return newResults[i].Target < newResults[j].Target
+	})
+	return newResults
```

Comment:
nit: We can probably remove an iteration over `vulns`.

```suggestion
	// Save packages to display them in the table even if no vulnerabilities were found
	resultMap := lo.SliceToMap(result.Packages, func(pkg ftypes.Package) (string, *types.Result) {
		return rootJarFromPath(pkg.FilePath), &types.Result{
			Target: lo.Ternary(pkg.FilePath != "", pkg.FilePath, result.Target),
			Class:  result.Class,
			Type:   result.Type,
		}
	})

	for _, vuln := range result.Vulnerabilities {
		pkgPath := rootJarFromPath(vuln.PkgPath)
		resultMap[pkgPath].Vulnerabilities = append(resultMap[pkgPath].Vulnerabilities, vuln)
	}

	newResults := lo.Values(resultMap)
	sort.Slice(newResults, func(i, j int) bool {
		return newResults[i].Target < newResults[j].Target
	})
	return lo.FromSlicePtr(newResults)
```

Reply:
nice catch!
Removed in https://github.com/aquasecurity/trivy/pull/8177/commits/ba8f75697d971da648ace136657f323160d9fcce